semgrep/semgrep/semgrep_main.py

```diff
@@ -168,6 +167,7 @@ def get_config(
     # first check if user asked to generate a config
     if generate_config:
         semgrep.config_resolver.generate_config()
+        sys.exit(0)
```

Comment:
Let's continue the trend of https://github.com/returntocorp/semgrep/pull/916 and try to limit number of places we call exit. Can probably move this up the call chain

Reply:
This is preserving existing behavior/making mypy happy

Reply:
Yeah, this was one of the `sys.exit` calls that was more difficult to pull out. Shouldn't be too much work - I would just pull the `if generate_config` check into the `cli.cli()` function, call `config_resolver.generate_config()` from there, and do an early `return`. But no worries if that seems like too much here.

Reply:
Oh i see. sg.